forem/forem
# cache open graph response

## What type of PR is this? (check all applicable)


## Description

This PR enhances the UX for authors by decreasing load time by caching the HTML response when an author embeds a url that uses OpenGraph. The OpenGraph data is then parsed from the cached HTML. Current cache expiration is set to 15 minutes.

**Note:** This was put together pretty quick and probably only a small percentage of folks will benefit initially. That said, the thought here is to build upon this as things grow and change.

## Related Tickets & Documents

- Closes #17005 

## QA Instructions, Screenshots, Recordings

- Draft a post and embed a "non-supported" url such as `https://weather.com`
- Open Inspector and view network tab (may have to check "preserve log")
- Click "Preview" tab
- Take note of the time it takes to load
- Flip back to Edit, then click Preview again
- Check the time it takes to load; should be a lot faster

### UI accessibility concerns?

No

## Added/updated tests?

- [x] No, and this is why: This is a private method and it's a core piece of functionality (`Rails.cache.fetch`) that's being used everywhere. I'm not sure what a test here would gain us considering it's just writing/fetching a cache key. I can be convinced otherwise.

## [Forem core team only] How will this change be communicated?
